### ISSUE TITLE
Dir.chdir not working for prepare_command when run via `pod lib lint`

### ISSUE THREAD

**Issue by arifken** [CocoaPods/CocoaPods#5873]
Tl;dr- seems related to this:  https://github.com/CocoaPods/Xcodeproj/pull/404. See the bottom of this description for more info

* [X] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?
I ran `pod lib lint` on my person pod with a spec that contains the Charts pod as a dependency. The Charts pod has a `prepare_command` that references files relative to its PODS_ROOT

## What did you expect to happen?
I expected the prepare_command to complete successfully and find the files it was affecting via `sed`

## What happened instead?
The prepare_command for Charts throws an error because it is clear that the script in prepare_command is not being run in the working directory for Charts-- it's being run in the working directory for my pod (that contains Charts as a dependency).

### Stack

Debugging using the master branch (via Rainforest), but reproducible in RC1 as well. 

```
   CocoaPods : 1.1.0.rc.2
        Ruby : ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin15]
    RubyGems : 2.5.1
        Host : Mac OS X 10.11.6 (15G1004)
       Xcode : 8.0 (8A218a)
         Git : git version 2.7.0
Ruby lib dir : /Users/mac/.rvm/rubies/ruby-2.3.0/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 547660e61a96b40acce78d1f125cdb4d1d56f1c3
               [two more private specs repos here]
```

### Installation Source

```
Executable Path: /Users/mac/.rvm/gems/ruby-2.3.0/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.1.0
slather               : 2.2.1
```

## Project that demonstrates the issue

I don't have an example, but I did some investigating, and here's what I found:

- I think the root cause stems from this issue: https://github.com/CocoaPods/Xcodeproj/pull/404
If i copy the override for Dir.chdir suggested in this PR and put that at the top of pod_source_preparer.rb, this problem is solved. 

- The bug is that something in Xcodeproj.PBXProject is causing Dir.chdir to break CocoaPods core. So what happens in in pod_source_preparer.rb is that the call to Dir.chdir on line 90 (`Dir.chdir(path) do ...`) does not have any effect, and if you log Dir.pwd inside that block it is set to the original working directory, not the one specified in the block. 

Any ideas?

**Comment by DanToml** [comment]
Thanks! - This should have been fixed by https://github.com/CocoaPods/Xcodeproj/pull/415/files :/

Can you try running with `FORK_XCODE_WRITING=true`?

**Comment by arifken** [comment]
oh awesome, ok let me take a look and see if the branch i'm working on has that code. thanks

**Comment by arifken** [comment]
@DanToml Confirmed that FORK_XCODE_WRITING fixes the issue- are there plans to have that turned on by default?

**Comment by PetresS** [comment]
Can someone tell me where to put `FORK_XCODE_WRITING=true` ?

i tryed with: 
`FORK_XCODE_WRITING=true pod repo push cocoapodsSpecRepo test.podspec --allow-warnings`

but nothing changed :(

**Comment by DanToml** [comment]
@arifken Unfortunately forking the process is not really a good default for this right now. We are however actively working on a more stable long term solution for handling the reading and writing of Xcode projects.

**Comment by arifken** [comment]
👍 Thanks @DanToml. Any chance you could reference this issue when the long term solution is integrated? Just so we know when we can stop using FORK_XCODE_WRITING :)
